test/blackbox-tests/test-cases/intf-only/excludes-by-modules-field.t

```diff
@@ -0,0 +1,27 @@
+Specifying a module without implementation that isn't inside the (modules ..)
+field
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.7)
+  > EOF
+  $ cat >dune <<EOF
+  > (library
+  >  (name foo)
+  >  (wrapped false)
+  >  (modules_without_implementation x)
+  >  (modules y))
+  > EOF
+
+  $ touch x.mli
+
+  $ cat >y.ml <<EOF
+  > module type F = X
+  > EOF
+
+  $ dune build --display short
```

Comment:
The behaviour we desire is for Dune to have errored earlier right?

Reply:
Yeah